backend/app.py

```diff
@@ -1,12 +1,24 @@
-from flask import Flask
-
-app = Flask(__name__)
+import requests
+from flask import Flask, jsonify
 
+def parse(forecast):
+    parsed = []
+    for period in forecast['periods']:
+        if period['name'] == 'Tonight':
+            parsed.append(period['name'])
+            parsed.append(period['detailedForecast'])
+        if period['name'] == 'Rest Of Today':
+            parsed.append(period['name'])
+            parsed.append(period['detailedForecast'])
+    return parsed
```

Comment:
The parsed output shouldn't be parsed as one long list; Instead, use a dictionary and map a key to a value, so that you get a structure that can be easily used by the frontend, and not rely on arbitrary positioning.

Something like:

```json
{
    "today": {"high": 54, "low": 32},
    "daily": {
        "0":  {"temp": 32, "weather": "normal"},
        "1":  {"temp": 35, "weather": "rain"},
        "2":  {"temp": 42, "weather": "snow"},
    },
    "weekly": {
        "Mon":  {"high": 54, "low": 32} 
    }
}
```